machine.go

```diff
@@ -55,9 +56,25 @@ type Machine struct {
 	scratchdev  string
 }
 
+// Create a new machine object with the auto backend
+func NewMachine() *Machine {
+	m, err := NewMachineWithBackend("auto")
+	if err != nil {
+		panic(err)
+	}
+	return m
+}
+
 // Create a new machine object
-func NewMachine() (m *Machine) {
-	m = &Machine{memory: 2048, numcpus: runtime.NumCPU()}
+func NewMachineWithBackend(backend string) (*Machine, error) {
+	m := &Machine{backend: backend, memory: 2048, numcpus: runtime.NumCPU()}
+
+	// check the backend is supported
+	_, err := newBackend(*m)
```

Comment:
This is weird; why would you create a backend just to test it's supported.. And why does the machine have a backend string. 

Why not create the backend here; save the pointer in the Machine struct and call it from that later.